### PR TITLE
Add basic parse tests

### LINKED ISSUE #5: Write tests
We should add a few tests. Basically: pass a query an verify that the proper objects with proper properties where instantiated. Also a wrong query should be pass and verify the Exception thrown.

### PR DESCRIPTION
**Pull request by merkkp**
I'd like some feedback on this :)
Closes #5 (maybe partially cause I don't check for the statement properties for example).

### CODE REVIEW

File: test/test_parse.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    def test_has_insert_and_select_statement(self):
+        query = "INSERT INTO my_table(id) VALUES(1); SELECT * FROM my_table"
+        insert_and_select_stmt = psqlparse.parse(query)
+        stmt_types = [stmt.type for stmt in insert_and_select_stmt]
+        self.assertListEqual(stmt_types, ['InsertStmt', 'SelectStmt'])
+
+    def test_syntax_error_select_statement(self):
+        query = "SELECT * FRO my_table"
+        try:
+            psqlparse.parse(query)
```
Comment (alculquicondor) [inline test/test_parse.py]: add `self.fail` here

File: test/test_parse.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        except Exception as e:
+            exception = e
+        self.assertIsInstance(exception, psqlparse.exceptions.PSqlParseError)
```
Comment (alculquicondor) [inline test/test_parse.py]: check for character position

File: test/test_parse.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+import psqlparse
+
+
+class TestParse(unittest.TestCase):
+
+    def test_is_select_statement(self):
+        query = "SELECT * FROM my_table"
+        select_stmt = psqlparse.parse(query).pop()
+        self.assertEqual(select_stmt.type, "SelectStmt")
```
Comment (alculquicondor) [inline test/test_parse.py]: I would like a few more checks. Like `with_clause` not set, `target_list` set and so on. Same for next test.

**Comment by merkkp** [comment]
Thinking about test cases are hard, even more for beginners
Thanks for the patience
Is it on the right track?

File: test/test_parse.py (hunk 3 above)
Comment (alculquicondor) [inline test/test_parse.py]: yes :)

File: test/test_parse.py (hunk 2 above)
Comment (alculquicondor) [inline test/test_parse.py]: There's a property on the exception object that accounts for the character position where the first error was found

File: test/test_parse.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        stmt_types = [stmt.type for stmt in insert_and_stmt]
+        self.assertListEqual(['InsertStmt', 'SelectStmt'], stmt_types)
+
+    @unittest.expectedFailure
+    def test_syntax_error_select_statement(self):
+        query = "SELECT * FRO my_table"
+        try:
+            parse(query)
+        except PSqlParseError as e:
+            self.fail()
```
Comment (alculquicondor) [inline test/test_parse.py]: sorry, what I meant is something like this:
```
try:
    parse(query)
    self.fail('nice failure  message')
except PsqlParseError as e:
     # now check e properties :)

File: test/test_parse.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        self.assertIsInstance(stmt.where_clause, WhereClause)
+        target = stmt.target_list.targets.pop()
+        self.assertDictEqual({'str': 'col1'},
+                             target['val']['ColumnRef']['fields'][0]['String'])
+        self.assertDictEqual({'str': '~~'},
+                             stmt.where_clause.obj['A_Expr']['name'][0]['String'])
+
+    def test_select_join(self):
+        query = "SELECT * FROM table_one JOIN table_two USING (common)"
+        stmt = parse(query).pop()
```
Comment (alculquicondor) [inline test/test_parse.py]: This failed... I guess it is an error in the library actually, isn't it?

Comment (merkkp) [inline test/test_parse.py]: Yep, it is. I think it is being raised by any join statement, should it be skipped it?

File: test/test_parse.py (hunk 4 above)
Comment (merkkp) [inline test/test_parse.py]: ohh, I get it, nice :)

File: test/test_parse.py (hunk 5 above)
Comment (alculquicondor) [inline test/test_parse.py]: No, I'll fix the library.

Comment (alculquicondor) [inline test/test_parse.py]: See.... tests :)

**Comment by merkkp** [comment]
Later I'll think about more cases :)

**Comment by alculquicondor** [comment]
I'm thinking that we need to implement #6 to properly solve this :/

**Comment by alculquicondor** [comment]
@merkkp please merge master into your branch

**Comment by merkkp** [comment]
Done!
And it is passing green. Nice!

**Pull request merged**